pkg/machine/apple/vfkit/helper.go

```diff
@@ -83,7 +83,9 @@ func (vf *Helper) stateChange(newState rest.StateChange) error {
 	}
 	payload := bytes.NewReader(b)
 	serverResponse, err := vf.post(vf.Endpoint+state, payload)
-	_ = serverResponse.Body.Close()
+	if err == nil {
+		_ = serverResponse.Body.Close()
```

Comment:
do you need to check if serverResponse is nil and/or the Body?

Reply:
I don't think so, unless we have to check this everywhere.  v.get() as well.

Reply:
if vf.post() returns an error then serverResponse is nil.  I would think  client.Do(req) will return the correct serverResponse or nil with an err.

Reply:
yeah checking the error alone should be fine